lldb/source/Plugins/Process/Utility/ThreadMemory.h

```diff
@@ -72,12 +72,17 @@ class ThreadMemory : public lldb_private::Thread {
 
   void ClearStackFrames() override;
 
-  void ClearBackingThread() override { m_backing_thread_sp.reset(); }
+  void ClearBackingThread() override {
+    if (m_backing_thread_sp)
+      m_backing_thread_sp->ClearBackedThread();
+    m_backing_thread_sp.reset();
+  }
```

Comment:
I really wanted to make `ClearBackedThread` a protected member of Thread, but then we can't access it here. Protected only works if you're going through `this`.